server/pulp/plugins/rsync/publish.py

```diff
@@ -0,0 +1,522 @@
+        """
+        start_date = config.get(START_DATE_KEYWORD)
+        end_date = config.get(END_DATE_KEYWORD)
+        date_filter = None
+        if start_date and not isinstance(start_date, basestring):
+            start_date_str = dateutils.format_iso8601_datetime(start_date)
+        else:
+            start_date_str = start_date
+        if end_date and not isinstance(end_date, basestring):
+            end_date_str = dateutils.format_iso8601_datetime(end_date)
+        else:
+            end_date_str = end_date
+
+        if start_date and end_date:
+            date_filter = {ASSOCIATED_UNIT_DATE_KEYWORD: {"$gte": start_date_str,
+                                                          "$lte": end_date_str}}
+        elif start_date:
+            date_filter = {ASSOCIATED_UNIT_DATE_KEYWORD: {"$gte": start_date_str}}
+        elif end_date:
+            date_filter = {ASSOCIATED_UNIT_DATE_KEYWORD: {"$lte": end_date_str}}
+        return date_filter
+
+    def get_remote_repo_path(self):
+        """
+        Returns the full path to the published repository on remote server.
+        """
+        if "relative_url" in self.repo.notes and self.repo.notes["relative_url"]:
+            return self.repo.notes["relative_url"]
+        else:
+            return self.repo.id
+
+    def get_repo_content_types(self):
+        """
+        This method needs to be implemented in each plugin. It returns a list of content types
+        supported by the plugin.
+        """
+        raise NotImplementedError()
+
+    def get_master_directories(self):
+        """
+        Returns a list of paths to master directories of the predistributor.
+
+        :return: list of absolute paths
+        :rtype: list of strings
+        """
+        dirs = []
+        for x in self.get_repo_content_types():
+            repo_relative_path = self.predistributor['config']['relative_url']
+            dirs.append(os.path.realpath(os.path.join(self._get_root_publish_dir(),
+                                                      repo_relative_path)))
+        return dirs
+
+    def get_units_directory_dest_prefix(self):
+        """
+        Returns the path on the remote server where content units should be stored.
+
+        :return: absolute path to /var/lib/pulp/
+        """
+
+        if self.get_config().get("remote_rel_prefix"):
+            origin_dest_prefix = self.get_config().get("remote_rel_prefix")
+        elif len(self.get_config().get("content_types")) >= 2:
+            origin_dest_prefix = os.path.join("content", "units")
+        else:
+            origin_dest_prefix = os.path.join("content", "units",
+                                              self.get_config().get("content_types")[0])
+
+        return origin_dest_prefix
+
+    def get_original_list(self):
+        """
+        Returns a list of files that need to be rsynced to the content directory
+
+        :return: List of files from /var/lib/pulp/content that need to be rsynced
```

Comment:
File names only? Or relative paths? If the latter, relative to what? In any case, it should be specified. Saying just "files" could lead someone to think they are file objects. Instead, I think you want some portion of the paths to files.